src/pageServices/contentScripts/highlight.js

```diff
@@ -241,9 +255,16 @@ export const highlightOnPage = () => {
       if (isInViewport(element) && !isHiddenByOverflow(element)) {
         const hash = element.getAttribute("jdn-hash");
         const highlightElement = document.getElementById(hash);
+        const predicted = predictedElements.find((e) => e.jdnHash === hash);
         if (!highlightElement) {
-          const predicted = predictedElements.find((e) => e.jdnHash === hash);
           drawRectangle(element, predicted);
+        } else {
+          const elementRect = element.getBoundingClientRect();
+          const highlightElementRect = highlightElement.getBoundingClientRect();
+          if (JSON.stringify(elementRect) !== JSON.stringify(highlightElementRect)) {
+            highlightElement.remove();
+            drawRectangle(element, predicted);
+          }
```

Comment:
let's refact multiple calls of `getBoundingClientRect()`. 
Problem: in `findAndHighlight` we call it in many places for same element: inside `isInViewport`, `isHiddenByOverflow`, `drawRectangle` and `else` block`. 
Solution: let's optimize it and call it once, at `findAndHighlight` and pass to all other functions.